### DISCUSSION TITLE
Windows Defender flagging Irony (again)?

### DISCUSSION THREAD

**Discussion by bcssov** [bcssov/IronyModManager#257] [General]
It's a false positive in an external component that Irony is using and it affects only versions <=1.16.17 which use this falsely flagged version. In this case you have 2 options:
1. Whitelist Irony temporarily and run autoupdate and update to >=1.16.25
2. Manually download 1.16.25 or later

If you want to learn more about what has transpired see [here](https://github.com/bcssov/IronyModManager/issues/255).

If Irony is still being flagged, open cmd (run as admin) and run the following commands. These will force update windows defender signature updates.

```
cd "%ProgramFiles%\Windows Defender"
MpCmdRun.exe -removedefinitions -dynamicsignatures
MpCmdRun.exe -SignatureUpdate
```

**Comment by bcssov** [comment]
It seems like again a false positive detection in the latest Avalonia binaries is occurring. It should clear up eventually once the file(s) are submitted to MS for analysis. Until meantime whitelist the offending DLLs or wait until signatures are updated and Defender whitelists the binaries once more.

**Comment by bcssov** [comment]
The issue is supposedly resolved, you can try to force update defender definitions:
```
cd "%ProgramFiles%\Windows Defender"
MpCmdRun.exe -removedefinitions -dynamicsignatures
MpCmdRun.exe -SignatureUpdate
```

**Comment by ghost** [comment]
noticed heavy throttling by antimalware process yesterday but whitelisting didnt help for whatever reason, may need to regedit a temp fix - appreciate the update posts on this issue even though it is entirely out of your hands

**Reply by bcssov** [reply]
Only windows defender is causing these issues (for every app that is using Avalonia framework that is). Since you didn't mention the type of software you use I'm going to assume it's windows defender. In your case:
1. Run the following command line commands (steps provided by Microsoft Windows Defender team)
2. Reinstall Irony
